client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+    useState,
+    useEffect 
+} from "react";
+import {useNavigate} from "react-router-dom";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText"
+import DialogActions from "@mui/material/DialogActions";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Link from "@mui/material/Link";
+import Tooltip from "@mui/material/Tooltip";
+import Fab from "@mui/material/Fab";
+import SendIcon from "@mui/icons-material/Send";
+import { 
+    usePostZoopMutation, 
+    useGetAllUsersQuery,
+    useGetMeQuery
+}  from "../features/api";
+import { useSelector } from "react-redux";
+import { RootState} from "../app/store";
+import type { User } from '../../src/types/custom'
+
+const SendZoopDialog = () => {
+    const [open, setOpen] = useState(false);
+    const [username, setUsername] = useState("");
+    const [content, setContent] = useState("");
+    const [zoopSent, setZoopSent] = useState(false);
+    const [lastZoopSentId, setLastZoopSentId] = useState<number | null>(null);
+
+    // Variable used for authorization
+    const token = useSelector((state: RootState) => state.auth.token);
+
+    const currentUser = useGetMeQuery().data;
+
+    const [postZoop, { isLoading: isPostZoopLoading, isError, data: zoopData }] = usePostZoopMutation();
+    const {data: usersData, isLoading: isGetAllUsersLoading, error } = useGetAllUsersQuery() 
+
+    const navigate = useNavigate();
+
+    const handleSendZoopClick = () => {
+        if (!usersData) {
+            throw new Error('No users to send Zoop to');
+            // TODO: How best to render a message about this error in UI?
+        } else {
+            const [recipient] = usersData.users.filter(user => user.username === username);
+            if (recipient) {
+                postZoop({
+                    content: content,
+                    authorId: currentUser?.user.id,
+                    receiverId: recipient.id
+                })
+                .unwrap()
+                .then((payload) => {
+                    console.log('fulfilled', payload);
+                    setLastZoopSentId(payload.zoop.id)
+                })
+                .catch((error) => console.error('rejected', error));
+              
+                setZoopSent(true)
+            } else {
+                throw new Error('No such user exists. Please select a different username');
+                // TODO: How best to render this error in UI?
+            }
+            
+        }
+        
+    }
+
+    return ( //TODO: MUI suggests adding a wrapper element so that Tooltip works when Fab is disabled
+        <>
+            <Tooltip 
+                title="Send a Zoop"
+            >
```

Comment:
Prefer explicit text in the button to a Tooltip in cases like this. Tooltips aren't really meant for situations like this, it's a little hard to explain, but it mostly comes down to - once you know what it is, you shouldn't have to see this tooltip every time. Tooltips cause motion which is visually distracting. Save Tooltips for things like form fields with an optional tooltip someone can use to learn more about a term.